src/character.cpp

```diff
@@ -2001,3 +2001,36 @@ bool Character::pour_into( vehicle &veh, item &liquid )
     }
     return true;
 }
+
+long Character::ammo_count_for( const item &gun )
+{
+    // INT_MAX instead of LONG_MAX because some code may not expect > INT_MAX
+    long ret = INT_MAX;
+    if( !gun.is_gun() ) {
+        return ret;
+    }
+
+    long required = gun.ammo_required();
+
+    if( required > 0 ) {
+        long remaining = 0;
+        if( gun.magazine_integral() ) {
```

Comment:
Kinda forgot what I was doing here, I guess.

I want this section to:
* Always add `ammo_remaining`
* Only find nearby ammo if at least one usable magazine is found
* Get the total number of shots that can be fired with ammo in gun + usable ammo nearby (if any)